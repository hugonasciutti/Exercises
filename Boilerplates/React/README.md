# Features

Dry boiler plate with:

- Webpack dev server
- React
- Babel Env
