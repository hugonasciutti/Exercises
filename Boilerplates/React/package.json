{
  "name": "boilerplatereact",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "webpack-dev-server"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "final-form": "^4.5.2",
    "react": "^16.3.2",
    "react-dom": "^16.3.2"
  },
  "devDependencies": {
    "babel-core": "^6.26.0",
    "babel-loader": "^7.1.4",
    "babel-preset-env": "^1.6.1",
    "babel-preset-react": "^6.24.1",
    "babel-preset-stage-0": "^6.24.1",
    "html-webpack-plugin": "^3.2.0",
    "webpack": "^4.6.0",
    "webpack-cli": "^2.0.15",
    "webpack-dev-server": "^3.1.3"
  }
}
