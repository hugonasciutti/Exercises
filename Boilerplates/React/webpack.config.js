const HtmlWebpackPlugin = require('html-webpack-plugin')

module.exports = {
  mode: 'development',

  entry: './src/index.jsx',

  output: {
    path: `${__dirname}/dist`,
    filename: 'app.js'
  },

  devtool: 'source-map',

  module: {
    rules: [
      { test: /\.(js|jsx)?$/, loader: 'babel-loader', exclude: /node_modules/ }
    ]
  },
  plugins: [
    new HtmlWebpackPlugin({
      title: 'Boilerplate',
      template: './src/index.ejs'
    })
  ]
}
