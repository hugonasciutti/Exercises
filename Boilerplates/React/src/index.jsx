import React, { Component } from 'react'
import { render } from 'react-dom'
import Form from './Form'

class App extends Component {
  render() {
    return (
      <div>
        <Input />
      </div>
    )
  }
}

render(<App />, document.getElementById('app'))