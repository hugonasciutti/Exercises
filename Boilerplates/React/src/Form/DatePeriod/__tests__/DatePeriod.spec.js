import React from 'react'
import { shallow } from 'enzyme'
import moment from 'core/utils/moment'
import '../../../../unit-test/fixture/datepicker'
import { DatePeriod } from '../'

describe('Date Period Component', () => {
  it('should render', () => {
    const component = shallow(<DatePeriod value={undefined} onChange={jest.fn()} />)

    expect(component).toMatchSnapshot()
  })

  it('should render and replace date format', () => {
    const component = shallow(<DatePeriod value={undefined} onChange={jest.fn()} dateFormat="MM/DD/Y" />)

    expect(component).toMatchSnapshot()
  })

  it('should change start date', () => {
    const startDate = moment()
    const value = { startDate: startDate.toISOString(), endDate: '' }
    const onChange = jest.fn()
    const component = shallow(<DatePeriod value={value} onChange={onChange} />)

    component.find('.rw-date-period-first').simulate('change', startDate)

    expect(onChange).toHaveBeenCalledWith(value)
  })

  it('should change end date', () => {
    const endDate = moment()
    const value = { startDate: '', endDate: endDate.toISOString() }
    const onChange = jest.fn()
    const component = shallow(<DatePeriod value={value} onChange={onChange} />)

    component.find('.rw-date-period-last').simulate('change', endDate)

    expect(onChange).toHaveBeenCalledWith(value)
  })
})
