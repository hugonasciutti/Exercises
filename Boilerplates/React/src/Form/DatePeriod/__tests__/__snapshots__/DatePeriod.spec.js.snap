// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Period Component should render 1`] = `
<div
  className="rw-date-period d-flex flex-row"
>
  <DatePicker
    className="rw-date-period-first form-control"
    dateFormat="MM/DD/YYYY"
    onChange={[Function]}
    required={false}
    selectsStart={true}
  />
  <span
    className="rw-date-period-separator d-flex align-items-center px-2"
  >
    <InjectIntl(Translate)
      id="component.date.period.separator"
    />
  </span>
  <DatePicker
    className="rw-date-period-last form-control"
    dateFormat="MM/DD/YYYY"
    disabled={true}
    onChange={[Function]}
    required={false}
    selectsEnd={true}
  />
</div>
`;

exports[`Date Period Component should render and replace date format 1`] = `
<div
  className="rw-date-period d-flex flex-row"
>
  <DatePicker
    className="rw-date-period-first form-control"
    dateFormat="MM/DD/Y"
    onChange={[Function]}
    required={false}
    selectsStart={true}
  />
  <span
    className="rw-date-period-separator d-flex align-items-center px-2"
  >
    <InjectIntl(Translate)
      id="component.date.period.separator"
    />
  </span>
  <DatePicker
    className="rw-date-period-last form-control"
    dateFormat="MM/DD/Y"
    disabled={true}
    onChange={[Function]}
    required={false}
    selectsEnd={true}
  />
</div>
`;
